### PR TITLE
check for expiration time of certificate. If expired, do not accept it

### PR DESCRIPTION
**Pull request by adelowo**
Fixes https://github.com/TykTechnologies/tyk-analytics/issues/1638 and https://github.com/TykTechnologies/tyk-analytics/issues/1637

> The certificate used in the test was generated by @maciejwojciechowski in https://github.com/TykTechnologies/tyk-analytics/issues/1638

### CODE REVIEW

**Comment by adelowo** [comment]
Should add a test with an expired certificate

File: certs/manager_test.go [hunk 1]
```diff
@@ -121,17 +122,30 @@ func TestAddCertificate(t *testing.T) {
 	cert2Raw, _ := pem.Decode(cert2Pem)
 	cert2ID := HexSHA256(cert2Raw.Bytes)
 
+	buf := new(bytes.Buffer)
+
+	f, err := os.Open("testdata/expired-cert.pem")
```
Comment (buger) [inline certs/manager_test.go]: Needs be replaced by dynamically generated certificate. See `genCertificate` usage in cert package.

Comment (adelowo) [reply certs/manager_test.go]: Done in https://github.com/TykTechnologies/tyk/pull/2712/commits/d84230485dae20592cd11acf67473d7bbe89cbee

**Pull request merged**